lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -644,4 +670,153 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
+
+                if( ev.getNewName().equals(latestName) )
+                    return ev;
+                else {
+                    return StatusEvent.
+                            getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(), eventType).
+                            withNullableAccessGroupID(Integer.decode(accessGrpId)).
+                            withNullableObjectID(objectId).
+                            withNullableVersion(ev.getVersion().get()).
+                            withNullableNewName(latestName).build();
+                }
+            } catch (IOException ex) {
+                throw handleException(ex);
+            } catch (JsonClientException ex) {
+                throw handleException(ex);
+            }
+        }
+        // update a (UN)PUBLISH_ALL_VERSIONS event
+        if (eventType.equals(StatusEventType.PUBLISH_ALL_VERSIONS) ||
+                eventType.equals(StatusEventType.UNPUBLISH_ALL_VERSIONS )) {
+            try {
+                // get globalread permission value
+                final String isPublic = getWorkspaceInfo(Long.decode(accessGrpId)).getE7();
+
+                // n = no permissions = public access
+                Boolean latestIsPublic = (isPublic == "n") ? true: false;
+
+                if (latestIsPublic) {
+                    return StatusEvent.
+                            getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
+                                    StatusEventType.PUBLISH_ALL_VERSIONS).
+                            withNullableAccessGroupID(Integer.decode(accessGrpId)).
+                            withNullableObjectID(objectId).
+                            withNullableVersion(ev.getVersion().get()).
+                            withNullableisPublic(latestIsPublic).build();
+                } else {
+                    return StatusEvent.
+                            getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
+                                    StatusEventType.UNPUBLISH_ALL_VERSIONS).
+                            withNullableAccessGroupID(Integer.decode(accessGrpId)).
+                            withNullableObjectID(objectId).
+                            withNullableVersion(ev.getVersion().get()).
+                            withNullableisPublic(latestIsPublic).build();
+                }
+
+            } catch (IOException ex) {
+                throw handleException(ex);
+            } catch (JsonClientException ex) {
+                throw handleException(ex);
+            }
+        }
+        // update a (UN)DELETE_ALL_VERSIONS event
+        if (eventType.equals(StatusEventType.DELETE_ALL_VERSIONS) ||
+                eventType.equals(StatusEventType.UNDELETE_ALL_VERSIONS )) {
+            Map<String, Object> command;
+
+            // if source data object does not exist then update event as a
+            // DELETE_ALL_VERSIONS event
+            command = new HashMap<>();
```

Comment:
So this is going to list the first 10000 objects in the workspace, which takes a bit of time, but more importantly, may not include the object of interest. Check out the max_obj_id and min_obj_id params to just list the one object you care about.

Reply:
Added min/max to call.